### PR TITLE
Rename cluster column pending to state

### PR DESCRIPTION
**Pull request by monstermunchkin**
Signed-off-by: Thomas Hipp <thomas.hipp@canonical.com>

### CODE REVIEW

**Comment by stgraber** [comment]
@monstermunchkin renamed the PR to only mention the column rename and associated changes.
This PR should just handle the rename and replacement in existing queries, not add anything about the evacuated state yet.

That's because the specification is still in draft state so we shouldn't be implementing parts of it yet :)

**Comment by stgraber** [comment]
But aligning the nodes table with what we have in storage and network is valuable on its own.

**Comment by monstermunchkin** [comment]
jenkins: test this please

File: lxd/db/cluster/query.go [hunk 1]
```diff
@@ -43,9 +43,12 @@ func selectNodesVersions(tx *sql.Tx) ([][2]int, error) {
 		return []interface{}{&versions[i][0], &versions[i][1]}
 	}
 
-	stmt, err := tx.Prepare("SELECT schema, api_extensions FROM nodes WHERE pending=0")
+	stmt, err := tx.Prepare("SELECT schema, api_extensions FROM nodes WHERE state=0")
 	if err != nil {
-		return nil, err
+		stmt, err = tx.Prepare("SELECT schema, api_extensions FROM nodes WHERE pending=0")
```
Comment (stgraber) [inline lxd/db/cluster/query.go:48]: @monstermunchkin can you add a comment there explaining why this is needed (it's clearly to make cluster updates work, but it'd be good to mention it still).

Comment (stgraber) [reply lxd/db/cluster/query.go:48]: I think we have something similar in the roles logic where we need to support the older schema.

**Comment by stgraber** [comment]
Looks good, just needs a comment to explain the DB fallback logic.

**Pull request merged**